server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -72,19 +74,53 @@ public MetaData getMetaData() {
     }
 
     public void start(Settings settings, TransportService transportService, ClusterService clusterService,
-                      MetaDataIndexUpgradeService metaDataIndexUpgradeService,
+                      MetaStateService metaStateService, MetaDataIndexUpgradeService metaDataIndexUpgradeService,
                       MetaDataUpgrader metaDataUpgrader, LucenePersistedStateFactory lucenePersistedStateFactory) {
         assert persistedState.get() == null : "should only start once, but already have " + persistedState.get();
 
         if (DiscoveryNode.isMasterNode(settings) || DiscoveryNode.isDataNode(settings)) {
             try {
-                persistedState.set(lucenePersistedStateFactory.loadPersistedState((version, metadata) ->
-                    prepareInitialClusterState(transportService, clusterService,
+                LucenePersistedStateFactory.OnDiskState onDiskState = lucenePersistedStateFactory.loadBestOnDiskState();
+
+                if (onDiskState == LucenePersistedStateFactory.NO_ON_DISK_STATE) {
+                    assert Version.CURRENT.major <= Version.V_7_0_0.major + 1 :
```

Comment:
We also return `LucenePersistedStateFactory.NO_ON_DISK_STATE` if starting up a brand-new node?

(I know this is merely moving code from an earlier PR, I missed this last time)

Reply:
Sorry, ignore this comment, there's not really an obvious better way to assert this.